mongo/src/main/scala/it/agilelab/darwin/connector/mongo/MongoConnector.scala

```diff
@@ -1,19 +1,131 @@
 package it.agilelab.darwin.connector.mongo
 
-import com.typesafe.config.Config
-import it.agilelab.darwin.common.Connector
+import com.mongodb.{BasicDBObject, ErrorCategory}
+import it.agilelab.darwin.common.{Connector, Logging}
+import it.agilelab.darwin.connector.mongo.ConfigurationMongoModels.BaseMongoConfig
 import org.apache.avro.Schema
+import org.apache.avro.Schema.Parser
+import org.mongodb.scala.bson.{BsonDocument, BsonValue}
+import org.mongodb.scala.{Document, MongoClient, MongoCollection, MongoWriteException, bson}
 
-class MongoConnector(config: Config) extends Connector {
-  override def fullLoad(): Seq[(Long, Schema)] = ???
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Await
+import scala.util.{Failure, Try}
 
-  override def insert(schemas: Seq[(Long, Schema)]): Unit = ???
+class MongoConnector(mongoClient: MongoClient, mongoConfig: BaseMongoConfig) extends Connector with Logging {
 
-  override def createTable(): Unit = ???
+  private def parser: Parser = new Parser()
 
-  override def tableExists(): Boolean = ???
+  override def fullLoad(): Seq[(Long, Schema)] = {
 
-  override def tableCreationHint(): String = ???
+    log.debug(s"loading all schemas from collection ${mongoConfig.collection}")
+    val collection =
+      mongoClient
+        .getDatabase(mongoConfig.database)
+        .getCollection(mongoConfig.collection)
 
-  override def findSchema(id: Long): Option[Schema] = ???
+    val schemas: Seq[Try[(Long, Schema)]] =
+      Await.result(
+        collection.find().map { document =>
+          for {
+            key <- extract(document, "_id", _.asInt64().getValue)
+            schemaStr <- extract(document, "schema", _.asString().getValue)
+            schema <- Try(parser.parse(schemaStr))
+          } yield key -> schema
+        }.toFuture(),
+        mongoConfig.timeout
+      )
+    log.debug(s"${schemas.size} loaded from MongoDB")
+    // this way the first exception is thrown, but we can change this line to support different error handling strategies
```

Comment:
This line is too long and fails scalastyle :(
```suggestion
    // this way the first exception is thrown, but we can change this line 
    // to support different error handling strategies
```